merkletree/pad.go

```diff
@@ -76,48 +93,65 @@ func (pad *PAD) Update(policies Policies) {
 		pad.loadedEpochs = append(pad.loadedEpochs[:0], pad.loadedEpochs[n:]...)
 	}
 
-	pad.updateInternal(policies, pad.currentSTR.epoch+1)
+	pad.updateInternal(policies, pad.latestSTR.epoch+1)
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	str := pad.latestSTR
+	if str == nil {
+		return ErrorSTRNotFound
+	}
+	index, _ := pad.computePrivateIndexInEpoch(key, str.epoch)
```

Comment:
This needs to use `pad.policies`, not the `latestSTR`.

There are two distinct times that we need to compute a private index,

* When setting a key.  This happens before a root is signed, and therefore cannot get the key from an STR.
* When looking up a key.  This always happens from a STR, and therefore uses the key from the policy in the STR.

So, it's fine to make `computePrivateIndex` a method of STR, but then you'll also need another method to handle the Set case.

Or, you can make it a method of PAD but one of the arguments needs to be a key.